src/presentation/device.rs

```diff
@@ -1,3 +1,12 @@
+//! This module is responsible for the device's interaction with the reader.
+//!
+//! It handles this through **State pattern**.
+//!
```

Comment:
I'm not sure this is a well-known pattern, so we should just describe it instead.
```suggestion
```

Reply:
maybe add the link for Wikipedia?

Reply:
I think it is okay to limit the State pattern to how the state pattern applies in this instance. For example:

```rust
//! This module is responsible for the estblishing the device's session with the reader.
//!
//! The device's [SessionManager] state machine is responsible for handling the session with the reader.
//!
//! The session is managed through a set of session management states: initalization, engaged, and established.
//!
//! To initalize a session management state, see the [SessionManagerInit] struct.
```

And then adding the relevant comments to the `SessionManager` structs. For Example:

```rust
/// The initial state of the Session Manager.
///
/// The Session Manager contains the documents, ephemeral device key, and device engagement.
///
/// Create a new Session Manager using the [SessionManagerInit::initialize] method, providing
/// the documents and optional non-empty list of device [DeviceRetrievalMethod] and
/// server [ServerRetrievalMethods] retrieval methods.
///
/// The [SessionManagerInit] state is restricted to creating a QR-code engagement,
/// using the [SessionManagerInit::qr_engagement] method, which will return the
/// [SessionManagerEngaged] Session Manager state.
///
/// For convience, the [SessionManagerInit] state surfaces the [SessionManagerInit::ble_ident] method
/// to provide the BLE identification string for the device.
#[derive(Serialize, Deserialize)]
pub struct SessionManagerInit {
    documents: Documents,
    e_device_key: Vec<u8>,
    device_engagement: Tag24<DeviceEngagement>,
}
```

Open to thoughts!

Reply:
thanks, adding those and adapting the other too